staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -297,11 +297,11 @@ func (j *JSONPath) findFieldInValue(value *reflect.Value, node *FieldNode) (refl
 	var inlineValue *reflect.Value
 	for ix := 0; ix < t.NumField(); ix++ {
 		f := t.Field(ix)
-		jsonTag := f.Tag.Get("json")
-		parts := strings.Split(jsonTag, ",")
-		if len(parts) == 0 {
+		jsonTag, found := f.Tag.Lookup("json")
```

Comment:
Just so I understand clearly, was the issue that splitting the empty string returned by f.Tag.Get when the "json" tag is unset results in a slice containing one element (`""`), and as a result the `len(parts) == 0` check failed to catch that case - leading to the belief that it's parsing an inline value, thus recursing into that struct? And the fix is to explicitly detect whether the "json" tag is set, because it's impossible to set this on inline values?

If that all sounds right then this lgtm.